### PR TITLE
[MySQL] Parse tables string to list of tables

### PR DESCRIPTION
**Pull request by timgrein**
## Closes https://github.com/elastic/enterprise-search-team/issues/4239

This PR adds a function to parse a MySQL tables string to a list of tables. It splits the string by comma. The regular expression also takes into account that a table name can contain a comma, if it's wrapped in backticks (possible in MySQL).

This change is needed for the new advanced rules structure coming in 8.8.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
~- [ ] Considered corresponding documentation changes~
~- [ ] Contributed any configuration settings changes to the configuration reference~

### CODE REVIEW

**Review by seanstory — DISMISSED** [review]
Gross. Does this need to be documented anywhere?

File: connectors/sources/mysql.py [hunk 1]
```diff
@@ -18,6 +19,8 @@
 from connectors.sources.generic_database import WILDCARD, configured_tables, is_wildcard
 from connectors.utils import CancellableSleeps, RetryStrategy, retryable
 
+SPLIT_BY_COMMA_OUTSIDE_BACKTICKS_PATTERN = r"`(?:[^`]|``)+`|\w+"
```
Comment (tarekziade) [inline connectors/sources/mysql.py:22]: not super important, but it's good practive to compile it:

```
SPLIT_TABLES = re.compile(r"`(?:[^`]|``)+`|\w+")
```

and then:

```
SPLIT_TABLES.findall(tables_string)
```

Comment (timgrein) [reply connectors/sources/mysql.py]: Right, thank you for the hint! Adjusted it.

**Comment by timgrein** [comment]
> Gross. Does this need to be documented anywhere?

@seanstory We can add it to the MySQL connector docs, that we support wrapping table names in backticks 👍

**Review by seanstory — APPROVED** [review]

**Pull request merged**